ext/native.go

```diff
@@ -208,13 +221,18 @@ func (tp *nativeTypeProvider) FindStructType(typeName string) (*types.Type, bool
 	return tp.baseProvider.FindStructType(typeName)
 }
 
-func toFieldName(parseStructTag bool, f reflect.StructField) string {
-	if !parseStructTag {
+func toFieldName(structTagToParse string, f reflect.StructField) string {
+	if structTagToParse == "" {
 		return f.Name
 	}
 
-	if name, found := f.Tag.Lookup("cel"); found {
-		return name
+	tag, found := f.Tag.Lookup(structTagToParse)
+	if found {
+		splits := strings.Split(tag, ",")
+		if len(splits) > 0 {
+			name := splits[0]
```

Comment:
Are we sure the name will always be in the first ordinal position on the tag split?

Reply:
Good question. I've added the following comment:

```
			// We make the assumption that the leftmost entry in the tag is the name.
			// This seems to be true for most tags that have the concept of a name/key, such as:
			// https://pkg.go.dev/encoding/xml#Marshal
			// https://pkg.go.dev/encoding/json#Marshal
			// https://pkg.go.dev/go.mongodb.org/mongo-driver/bson#hdr-Structs
			// https://pkg.go.dev/gopkg.in/yaml.v2#Marshal
```

As far as I can tell, the convention seems to be that the 0th entry is the name/key for tags that _have_ a name/key. Some don't have such a concept, such as https://pkg.go.dev/encoding/asn1, and using this option with `asn1` wouldn't make any sense because of that.

I quickly went through the list here: https://go.dev/wiki/Well-known-struct-tags - the one that jumped out at me as not following this rule was protobuf, which seems to do:
`protobuf:"bytes,1,opt,name=something,proto3"` (at least for proto3?), where the name could presumably be anywhere as long as it's annotated with the `name=` key within the tag.

I suppose if we really wanted to cater to cases like this, we could add to the `ParseStructTag(tag string)` option to look something like:
`func ParseStructTagDetailed(tag string, parser func(tag string) string) NativeTypesOption` where the user can define a tag-parsing function that could work for any arbitrary tag.

That feels like maybe an addition left to a later time if it's actually needed though, as it could be done in a nonbreaking way.

Reply:
Proto is handled as a special-case already, so I'm not too worried about it. However, let's document that the name is expected to be the 0th entry in the tag as part of the CEL option.

Reply:
> However, let's document that the name is expected to be the 0th entry in the tag as part of the CEL option.

Done. `ParseStructTag` now says:
```
// ParseStructTag configures the struct tag to parse. The 0th item in the tag is used as the name of the CEL field.
// For example: 
// If the tag to parse is "cel" and the struct field has tag cel:"foo", the CEL struct field will be "foo".
// If the tag to parse is "json" and the struct field has tag json:"foo,omitempty", the CEL struct field will be "foo".
```